### ISSUE TITLE
PBSCluster is not awaitable

### ISSUE THREAD

**Issue by aidanheerdegen** [dask/dask-jobqueue#273]
It would be great to use `PBSCluster` with `dask-labextension` but for this to work `PBSCluster` has to be able to be used asynchronously, so has to support `await`

https://github.com/dask/dask-labextension/pull/36

When I tried specifying PBSCluster in my `~/.dask/config`:
```yaml
labextension: 
  factory: 
    args: []
    class: PBSCluster
    module: dask_jobqueue
    kwargs: 
      cores: 8
      jobextra: 
        - "-P v45"
      memory: 16GB
      processes: 8
      queue: expressbw
      resource_spec: 'ncpus=8,mem=16G'
```
I got an error:
```
[W 13:05:24.228 LabApp] object PBSCluster can't be used in 'await' expression
```

**Comment by guillaumeeb** [comment]
That's quite possible that dask-jobqueue is not currently compatible. I have to admit my lack of knowledge with async programing. Work has been done with dask-kubernetes [here](https://github.com/dask/dask-kubernetes/pull/116) as you're probably aware through the dask-labextension issue.

Would you want to give a try on the equivalent for dask-jobqueue?

**Comment by aidanheerdegen** [comment]
I have attempted this, but no luck so far. I'll try and find time to work on it next week.

**Comment by guillaumeeb** [comment]
@mrocklin, should be fixed by #307 if I'm not mistaken?

**Comment by lesteve** [comment]
> @mrocklin, should be fixed by #307 if I'm not mistaken?

That's my understanding as well.

**Comment by mrocklin** [comment]
Yes

**Comment by danpf** [comment]
while you're waiting, if you wrap your `.submit` calls in `asyncio.ensure_future(` it should work.

**Comment by guillaumeeb** [comment]
Fixed by #307
